Sources/SwiftDocC/Model/MarkdownOutput/Translation/MarkdownOutputNodeTranslator.swift

```diff
@@ -0,0 +1,66 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2025 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+public import Foundation
+
+/// Creates ``CollectedMarkdownOutput`` from a ``DocumentationNode``.
+internal struct MarkdownOutputNodeTranslator {
+    
+    var visitor: MarkdownOutputSemanticVisitor
+    
+    init(context: DocumentationContext, node: DocumentationNode) {
+        self.visitor = MarkdownOutputSemanticVisitor(context: context, node: node)
+    }
+    
+    mutating func createOutput() -> CollectedMarkdownOutput? {
+        if let node = visitor.start() {
+            return CollectedMarkdownOutput(identifier: visitor.identifier, node: node, manifest: visitor.manifest)
+        }
+        return nil
+    }
+}
+
+struct CollectedMarkdownOutput {
+    let identifier: ResolvedTopicReference
+    let node: MarkdownOutputNode
+    let manifest: MarkdownOutputManifest?
+    
+    var writable: WritableMarkdownOutputNode {
+        get throws {
+            WritableMarkdownOutputNode(identifier: identifier, nodeData: try node.data)
+        }
+    }
+}
+
+@_spi(MarkdownOutput)
+public struct WritableMarkdownOutputNode {
+    public let identifier: ResolvedTopicReference
+    public let nodeData: Data
+}
+
+extension MarkdownOutputManifest {
+    var writable: WritableMarkdownOutputManifest {
+        get throws {
+            let encoder = JSONEncoder()
+            encoder.outputFormatting = [.sortedKeys, .withoutEscapingSlashes]
+            #if DEBUG
+            encoder.outputFormatting.insert(.prettyPrinted)
+            #endif
+            let data = try encoder.encode(self)
+            return WritableMarkdownOutputManifest(title: title, manifestData: data)
+        }
+    }
+}
+
+@_spi(MarkdownOutput)
+public struct WritableMarkdownOutputManifest {
+    public let title: String
+    public let manifestData: Data
```

Comment:
(same as above)
Question: what are clients/callers expected to do with this raw data? It has no documentation, that explain how a developer can use this structure. Is there a structured value that we could surface here instead? 

Alternatively, if there's no use case for accessing this node type, can we make it internal so that we keep the surface of new public SPI to a minimum?